### ISSUE TITLE
Random failing SorterTest.testSorterContentAddOverride

### ISSUE THREAD

**Issue by jukzi** [eclipse-platform/eclipse.platform.ui#809]
```
org.junit.ComparisonFailure: expected:<Blue[Parent]> but was:<Blue[src]>
	at org.eclipse.ui.tests.navigator.SorterTest.testSorterContentAddOverride(SorterTest.java:371)
```

for example
https://github.com/eclipse-platform/eclipse.platform.ui/pull/747#issuecomment-1538458088
linux:
https://github.com/eclipse-platform/eclipse.platform.ui/runs/14128127328

**Comment by jukzi** [comment]
also Random failing SorterTest.testComparatorAsSorter
```
expected:<Blue[Added]Parent> but was:<Blue[]Parent>
org.junit.ComparisonFailure: expected:<Blue[Added]Parent> but was:<Blue[]Parent>
	at org.eclipse.ui.tests.navigator.SorterTest.testComparatorAsSorter(SorterTest.java:315)
```
https://github.com/eclipse-platform/eclipse.platform.ui/actions/runs/5220601588/jobs/9424564254

**Comment by marcushoepfner** [comment]
I tried to look at it. First of all, locally when running in Eclipse as PDE test, there a lot of errors in the console. Same for [LabelProviderTest](https://github.com/eclipse-platform/eclipse.platform.ui/blob/master/tests/org.eclipse.ui.tests.navigator/src/org/eclipse/ui/tests/navigator/LabelProviderTest.java). Don't know if this is intented.

I have uploaded a first fix for that with #911 . But there are still errorr in the console. Some seems to be by intention, e.g. [LabelProviderTest.testLabelProviderThrow()](https://github.com/eclipse-platform/eclipse.platform.ui/blob/master/tests/org.eclipse.ui.tests.navigator/src/org/eclipse/ui/tests/navigator/LabelProviderTest.java#L429C14-L440).

**Comment by marcushoepfner** [comment]
Refactored [LabelProviderTest.testLabelProviderThrow()](https://github.com/eclipse-platform/eclipse.platform.ui/blob/master/tests/org.eclipse.ui.tests.navigator/src/org/eclipse/ui/tests/navigator/LabelProviderTest.java#L429C14-L440) as well.

waiting for the unfreeze, then I'll merge.
